### PR TITLE
compiler: fix few unused_peekable and needless_pass_by_ref_mut clippy lints

### PR DESCRIPTION
**Pull request by klensy**
This fixes few instances of `unused_peekable` and `needless_pass_by_ref_mut`. While i expected to fix more warnings, `needless_pass_by_ref_mut` produced too much for one PR, so i stopped here.

Better reviewed commit by commit, as fixes splitted by chunks.

### CODE REVIEW

File: compiler/rustc_borrowck/src/type_check/liveness/polonius.rs [hunk 1]
```diff
@@ -87,6 +87,8 @@ pub(super) fn populate_access_facts<'a, 'tcx>(
     body: &Body<'tcx>,
     location_table: &LocationTable,
     move_data: &MoveData<'tcx>,
+    //FIXME: this is not mutated, but expected to be modified as
+    // out param, bug?
     dropped_at: &mut Vec<(Local, Location)>,
```
Comment (klensy) [inline compiler/rustc_borrowck/src/type_check/liveness/polonius.rs:92]: This is called here
https://github.com/rust-lang/rust/blob/760e567af5398a0d8c512f904e551e1f38e00d79/compiler/rustc_borrowck/src/type_check/liveness/mod.rs#L55-L59
where drop_used expected to be populated, but it's not.

**Comment by klensy** [comment]
> @bors r+

Nice! What should i do about 2 added FIXMEs?

**Comment by Nilstrieb** [comment]
the `UnstableOptions` one can be fixed in a follow up PR, for https://github.com/rust-lang/rust/pull/123188/files#diff-31c8499e4394940eda9068f284ef5eccfd988bfe427168f6752d90a82dcc21cfR90 I'm not sure, maybe also try to figure out a solution and ask @lqd if you're not sure?

**Pull request merged**